nautobot_ssot/contrib.py

```diff
@@ -0,0 +1,354 @@
+                    is_custom_field = True
+                    continue
+            if is_custom_field:
+                continue
+
+            database_field = diffsync_model._model._meta.get_field(parameter_name)
+
+            # Handling of one- and many-to-many fields.
+            # Note: This includes the side of a generic foreign key that constitues the foreign key,
+            # i.e. the 'one' side.
+            if database_field.many_to_many or database_field.one_to_many:
+                parameters[parameter_name] = self._handle_to_many_relationship(
+                    database_object, diffsync_model, parameter_name
+                )
+                continue
+
+            # Handling of normal fields - as this is the default case, set the attribute directly.
+            parameters[parameter_name] = getattr(database_object, parameter_name)
+        try:
+            diffsync_model = diffsync_model(**parameters)
+        except pydantic.ValidationError as error:
+            raise ValueError(f"Parameters: {parameters}") from error
+        self.add(diffsync_model)
+
+        self._handle_children(database_object, diffsync_model)
+        return diffsync_model
+
+    def _handle_children(self, database_object, diffsync_model):
+        """Recurse through all the children for this model."""
+        for children_parameter, children_field in diffsync_model._children.items():
+            children = getattr(database_object, children_field).all()
+            diffsync_model_child = self._get_diffsync_class(model_name=children_parameter)
+            for child in children:
+                parameter_names = self._get_parameter_names(diffsync_model_child)
+                child_diffsync_object = self._load_single_object(child, diffsync_model_child, parameter_names)
+                diffsync_model.add_child(child_diffsync_object)
+
+    def load(self):
+        """Generic implementation of the load function."""
+        if not hasattr(self, "top_level") or not self.top_level:
+            raise ValueError("'top_level' needs to be set on the class.")
+
+        for model_name in self.top_level:
+            diffsync_model = self._get_diffsync_class(model_name)
+
+            # This function directly mutates the diffsync store, i.e. it will create and load the objects
+            # for this specific model class as well as its children without returning anything.
+            self._load_models(diffsync_model)
+
+    def _get_diffsync_class(self, model_name):
+        """Given a model name, return the diffsync class."""
+        try:
+            diffsync_model = getattr(self, model_name)
+        except AttributeError as error:
+            raise AttributeError(
+                f"Please define {model_name} to be the diffsync model on this adapter as a class level attribute."
+            ) from error
+        return diffsync_model
+
+    @staticmethod
+    def _handle_to_many_relationship(database_object, diffsync_model, parameter_name):
```

Comment:
you are also using "handle" in the `NautobotModel` methods. Maybe here we could use something like "get"? (because you are getting things)